pkg/patcher.go

```diff
@@ -0,0 +1,81 @@
+package pkg
+
+import (
+	"fmt"
+	"path"
+
+	csibaremetalv1 "github.com/dell/csi-baremetal-operator/api/v1"
+	"github.com/dell/csi-baremetal-operator/api/v1/components"
+)
+
+const (
+	rke2ManifestsFolder    = "/var/lib/rancher/rke2/agent/pod-manifests"
+	vanillaManifestsFolder = "/etc/kubernetes/manifests"
+
+	schedulerFolder = "scheduler"
+
+	policyFile   = "policy.yaml"
+	configFile   = "config.yaml"
+	config19File = "config-19.yaml"
+
+	policyPath   = schedulerFolder + "/" + policyFile
+	configPath   = schedulerFolder + "/" + configFile
+	config19Path = schedulerFolder + "/" + config19File
+)
+
+func NewPatcherConfiguration(csi *csibaremetalv1.Deployment) patcherConfiguration {
+	var config patcherConfiguration
+	switch csi.Spec.Platform {
+	case platformVanilla, "":
+		config = patcherConfiguration{
+			platform:        platformVanilla,
+			targetConfig:    path.Join(vanillaManifestsFolder, configPath),
+			targetPolicy:    path.Join(vanillaManifestsFolder, policyPath),
+			targetConfig19:  path.Join(vanillaManifestsFolder, config19Path),
+			schedulerFolder: path.Join(vanillaManifestsFolder, schedulerFolder),
+			manifestsFolder: vanillaManifestsFolder,
+		}
+	case platformRKE:
+		config = patcherConfiguration{
+			platform:        platformRKE,
+			targetConfig:    path.Join(rke2ManifestsFolder, configPath),
+			targetPolicy:    path.Join(rke2ManifestsFolder, policyPath),
+			targetConfig19:  path.Join(rke2ManifestsFolder, config19Path),
+			schedulerFolder: path.Join(rke2ManifestsFolder, schedulerFolder),
+			manifestsFolder: rke2ManifestsFolder,
+		}
+	default:
+		panic(fmt.Sprintf("Non supported platform %s", csi.Spec.Platform))
```

Comment:
Will panic crush the container? Isn't returning error better or have you special logic here?

Reply:
fixed